tmk_core/common/keyboard.c

```diff
@@ -336,32 +341,37 @@ void keyboard_task(void) {
     matrix_scan();
 #endif
 
-    if (should_process_keypress()) {
-        for (uint8_t r = 0; r < MATRIX_ROWS; r++) {
-            matrix_row    = matrix_get_row(r);
-            matrix_change = matrix_row ^ matrix_prev[r];
-            if (matrix_change) {
+    for (uint8_t r = 0; r < MATRIX_ROWS; r++) {
+        matrix_row    = matrix_get_row(r);
+        matrix_change = matrix_row ^ matrix_prev[r];
+        if (matrix_change) {
 #ifdef MATRIX_HAS_GHOST
-                if (has_ghost_in_row(r, matrix_row)) {
-                    continue;
-                }
+            if (has_ghost_in_row(r, matrix_row)) {
+                continue;
+            }
 #endif
-                if (debug_matrix) matrix_print();
-                matrix_row_t col_mask = 1;
-                for (uint8_t c = 0; c < MATRIX_COLS; c++, col_mask <<= 1) {
-                    if (matrix_change & col_mask) {
+            if (debug_matrix) matrix_print();
+            matrix_row_t col_mask = 1;
+            for (uint8_t c = 0; c < MATRIX_COLS; c++, col_mask <<= 1) {
+                if (matrix_change & col_mask) {
+                    if (should_process_keypress()) {
```

Comment:
I'm not a fan of moving this.  

The point of `should_process_keypress` is that it can be over-ridden.  And if we need to do so for any reason, it should be overwritten.  Eg, something like this, in split_util.c would make more sense, IMO:

```c
bool should_process_keypress(void) { 
    is_keyboard_master(); 
    return true;
}
```

And then below, replace `should_process_keypress()` with `is_keyboard_master()`. 

```suggestion
    if (should_process_keypress()) {
        for (uint8_t r = 0; r < MATRIX_ROWS; r++) {
            matrix_row    = matrix_get_row(r);
            matrix_change = matrix_row ^ matrix_prev[r];
            if (matrix_change) {
#ifdef MATRIX_HAS_GHOST
                if (has_ghost_in_row(r, matrix_row)) {
                    continue;
                }
#endif
                if (debug_matrix) matrix_print();
                matrix_row_t col_mask = 1;
                for (uint8_t c = 0; c < MATRIX_COLS; c++, col_mask <<= 1) {
                    if (matrix_change & col_mask) {
                         if (is_keyboard_master()) {

```

Reply:
The point of moving this is that this changed setup allows for systems that need to react to electrical matrix changes can do so without also triggering the entire key process loop. So should_process_keypress remains as overridable to allow key events to trigger, and that is independent from what rgb matrix needs to do.

Reply:
Well, in that case, wouldn't it make more sense to remove the ifdef for "oled task". and always have `matrix_scan`s status be dumped to a boolean?  And then check that, in the case for stuff like this. 

eg: 
```c
                            if (is_keyboard_master() 
#if defined(RGB_MATRIX_SPLIT)
                                || changed
#endif
                                  ) { 
```

Reply:
Not really, as in RGB Matrix's case it needs to react per key, not per matrix.

Reply:
The way I see how features need to react to changes in state basically falls into 4 categories: 
- Any change in state (OLED inactivity timer, RGB inactivity timer, etc)
- Any changes per key (Key reactive RGB effects)
- Any keycodes (process_***_task)
- Custom